code/script.js

```diff
@@ -303,41 +323,85 @@ const filterCharacters = (keep) => {
+     if (keep) {
+      alert(
+        `Yes, the person have ${value} eyes! Keep all that have ${value} eyes!`
+      )
+    } else {
+      alert(
+        `No, the person do not have ${value} eyes! Remove all that have ${value} eyes!`
+      )
+    }
+  }
+  
+  else if (category === 'other') {
+    if (keep) {
+      alert(
+        `Yes, the person smokes! Keep all persons that smoke`
+      )
+    } else {
+      alert(
+        `No, the person doesn't smoke! Remove all persons that smoke`
+      )
+    }
     // Similar to the one above
-  } else {
+  } 
+  else if (category === 'hair color') {
     if (keep) {
+      alert(`Yes, the person has ${value} hair! Keep all persons with ${value} hair`)
       // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
     } else {
+      alert(`No, the person doesn't have ${value} hair. Remove all persons with ${value} hair`)
       // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
-    }
+    } 
   }
 
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+ generateBoard()
 
   // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
+  const isConfirmed = confirm(`Are you sure you want to make a guess on ${suspect}?`)
+  if (isConfirmed) {
+    checkMyGuess(suspect)
+  }
   // store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
+  if (suspect === secret.name) {
+    winOrLoseText.innerHTML = `You guessed on ${suspect} and it was correct!`
+  } else {
+    winOrLoseText.innerHTML = `Wrong answer. It was actually ${secret.name}!`
+  }
+  console.log("suspect", suspect)
+  console.log("secret name", secret.name)
   // 1. Check if the suspect is the same as the secret person's name
   // 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
   // 4. Hide the game board
+  winOrLose.style.display = "flex"
+board.style.display = "none"
 }
 
+
 // Invokes the start function when website is loaded
 start()
 
 // All the event listeners
+playAgainButton.addEventListener('click', start)
 restartButton.addEventListener('click', start)
+questions.addEventListener('change', () => selectQuestion())
```

Comment:
I realize this is from the last Q/A but since we no longer pass along an argument, we can write this like the others:
`questions.addEventListener('change', selectQuestion)`